pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -452,7 +452,7 @@ func PrepareLogsQuery(start, end int64, queryType v3.QueryType, panelType v3.Pan
 		} else {
 			query = addLimitToQuery(query, mq.Limit)
 		}
-	} else if panelType == v3.PanelTypeTable {
+	} else if panelType == v3.PanelTypeTable && mq.Limit > 0 {
```

Comment:
I think Limit>0 condition can be added to `addLimitToQuery` function itself. This avoids repeated checks for limit>0 outside the function.